lib/ansible/modules/tacp/tacp_instance.py

```diff
@@ -288,30 +256,171 @@ def create_instance(instance_params):
             enable_automatic_recovery=instance_params['auto_recovery_enabled'],
             description=instance_params['description'])
 
-        result['api_request_body'] = str(body)
+        if module._verbosity >= 3:
+            result['api_request_body'] = str(body)
         try:
             # Create an application from a template
             api_response = api_instance.create_application_from_template_using_post(
                 body)
-            result['changed'] = True
-            result['failed'] = False
-            module.exit_json(**result)
-
+            if module._verbosity >= 3:
+                result['api_response'] = str(api_response)
         except ApiException as e:
 
             return "Exception when calling ApplicationsApi->create_application_from_template_using_post: %s\n" % e
 
-    create_instance_response = create_instance(instance_params)
+        tacp_utils.wait_for_action_to_complete(
+            api_response.action_uuid, api_client)
+        result['ansible_module_results'] = tacp_utils.get_resource_by_uuid(
+            api_response.object_uuid, 'application', api_client)
+        result['changed'] = True
+
+    def instance_power_action(name, api_client, action):
+        instance_uuid = tacp_utils.get_component_fields_by_name(
+            module.params['name'], 'application', api_client)
+        api_instance = tacp.ApplicationsApi(api_client)
+
+        actions = {"start": api_instance.start_application_using_put,
+                   "stop": api_instance.stop_application_using_put,
+                   "restart": api_instance.restart_application_using_put,
+                   "shutdown": api_instance.shutdown_application_using_put,
+                   "pause": api_instance.pause_application_using_put,
+                   "resume": api_instance.resume_application_using_put,
+                   "delete": api_instance.delete_application_using_delete}
+
+        try:
+            api_response = actions[action](
+                instance_uuid)
+            if module._verbosity >= 3:
+                result['api_response'] = str(api_response)
+        except ApiException as e:
+            response_dict = tacp_utils.api_response_to_dict(e)
+
+            if "Error" in response_dict['message']:
```

Comment:
When will "Error" not be in the response message but an ApiException is thrown? Why are we not failing for that case?

Reply:
No good reason, just an oversight :) Fixed here